extensions/amp-sidebar/0.1/test/integration/test-amp-sidebar.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {listenOncePromise} from '../../../../../src/event-helper';
+
+const config = describe.configure().ifNewChrome();
+
+config.run('amp-sidebar', function() {
+  this.timeout(100000);
+
+  const extensions = ['amp-sidebar'];
+
+  const sidebarBody = `
+  <amp-sidebar
+    id="sidebar1"
+    layout="nodisplay"
+    on="sidebarOpen:focusOnMe.focus">
+    <ul>
+      <li> <a id="focusOnMe" href="https://google.com">Focused on open</a></li>
+      <li>
+        <button id="sidebarCloser" on="tap:sidebar1.close">
+          Close
+        </button>
+      </li>
+    </ul>
+  </amp-sidebar>
+  <button id="sidebarOpener" on="tap:sidebar1.toggle">Open Sidebar</button>
+  <div id="section2" style="position: absolute; top: 1000px;">
+    <h1 >Section 2</h1>
+  </div>
+  <div id="section3" style="position: absolute; top: 2000px;">
+    <h1 >Section 3</h1>
+  </div>
+  `;
+  describes.integration('sidebar focus', {
+    body: sidebarBody,
+    extensions,
+  }, env => {
+
+    let win;
+    beforeEach(() => {
+      win = env.win;
+    });
+
+    it('should focus on opener on close', () => {
+      return new Promise((resolve, reject) => {
```

Comment:
You shouldn't need this outer promise, you can just return the `openedPromise` chain.